pubnub/event_engine/dispatcher.py

```diff
@@ -1,42 +1,42 @@
-from pubnub.event_engine.models import effects
-from pubnub.event_engine import manage_effects
+from pubnub.event_engine.models import invocations
+from pubnub.event_engine import effects
 
 
 class Dispatcher:
     _pubnub = None
-    _managed_effects_factory = None
+    _effects_factory = None
 
     def __init__(self, event_engine) -> None:
         self._event_engine = event_engine
         self._managed_effects = {}
-        self._effect_emitter = manage_effects.EmitEffect()
+        self._effect_emitter = effects.EmitEffect()
 
     def set_pn(self, pubnub_instance):
         self._pubnub = pubnub_instance
         self._effect_emitter.set_pn(pubnub_instance)
 
-    def dispatch_effect(self, effect: effects.PNEffect):
-        if not self._managed_effects_factory:
-            self._managed_effects_factory = manage_effects.ManagedEffectFactory(self._pubnub, self._event_engine)
+    def dispatch_effect(self, invocation: invocations.PNInvocation):
```

Comment:
Mhm, I don't think we should differentiate Effects that emit status (or anything else) from other Effects. They should have the same interface or method to fire them. I know it's not possible to fix it, so let's keep this comment so that others are aware of this difference.

Reply:
I will look at it